### ISSUE TITLE
the error about "initDatasources"

### ISSUE THREAD

**Issue by qwenwang** [fralx/LimeReport#48]
> for (int i=0;i<m_patternPageItem->dataBandCount() && !m_renderCanceled;i++){
        lastRenderedBand = m_patternPageItem->dataBandAt(i);
        initDatasources();
        renderDataBand(lastRenderedBand);
        if (i<m_patternPageItem->dataBandCount()-1) closeFooterGroup(lastRenderedBand);
    }
When rendering new dataBand, it will reset all the datasources and that will cause my program broken. I wrote 3 datasources, the second datasource is the subdatasource of the first datasource. when the program runs to render the third datasource, it will init all the datasources, and it init the second datasource first, and my program should init the detailIterator to the begin of mainIterator's child list, which cause the error, because the mainIterator was not initial and it point to the end of the main list.
I think it maybe better for only initializing the current datasource of the data band. 
Sorry for my poor english.

**Comment by fralx** [comment]
Sorry for long waiting. 
If you want to build report that contains nested data you should use subdetail band instead databand.

**Comment by qwenwang** [comment]
Thanks for your reply.
In fact, I used subdetail band. As I memtioned above, the first band is data band, and the second band is subdetail band, and the the third band is data band. 
When rendering the second databand(the third band), it will invoke initDatasources(). this function will init all the datasources I registered using createCallbackDatasouce function and it first emit first signal to my  subdetail datasource. But my subdetail datasource should not inited(), because my first data band has ended up and the iterator of the first databand's dataList equals to the end of the list, which cause the first() failed:
